### PR TITLE
Fix example code, clean up doc, and attempt to address customer feedback

### LINKED ISSUE #26774: Code block unclear
Hi,

First of all thank you very much for this very helpful walk-through!

The DecryptFile method contains this block:
```
 inFs.Seek(0, SeekOrigin.Begin);
 inFs.Seek(0, SeekOrigin.Begin);
 inFs.Read(LenK, 0, 3);
 inFs.Seek(4, SeekOrigin.Begin);
 inFs.Read(LenIV, 0, 3);
```
Questions:

- The first two lines are the same - why?
- Line three reads 3 bytes but isn't the size of the buffer 4 bytes long? Same applies for line five.

Then the line
`int lenC = (int)inFs.Length - startC;`
seems to be unnecessary?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2a8e441d-cd59-de15-a814-0d86d4cda052
* Version Independent ID: c94ca6d4-7e93-8aac-93d1-7f249c77d243
* Content: [Walkthrough: Creating a Cryptographic Application](https://docs.microsoft.com/en-us/dotnet/standard/security/walkthrough-creating-a-cryptographic-application)
* Content Source: [docs/standard/security/walkthrough-creating-a-cryptographic-application.md](https://github.com/dotnet/docs/blob/main/docs/standard/security/walkthrough-creating-a-cryptographic-application.md)
* Product: **dotnet**
* Technology: **dotnet-security**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Edit pass
- Clean up doc
- Fix code

Fixes #26774

✔️ [Internal Preview — Walkthrough: Creating a Cryptographic Application](https://review.docs.microsoft.com/en-us/dotnet/standard/security/walkthrough-creating-a-cryptographic-application?branch=pr-en-us-27189)

### CODE REVIEW

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 1]
```diff
@@ -1,223 +1,203 @@
...
-|<xref:System.Windows.Forms.OpenFileDialog>|`openFileDialog1`||  
-|<xref:System.Windows.Forms.OpenFileDialog>|`openFileDialog2`||  
-  
- Double-click the buttons in the Visual Studio designer to create their event handlers.
-  
-## Declaring Global Objects  
-
-Add the following code as part of the declaration of the class Form1. Edit the string variables for your environment and preferences.  
-  
+This walkthrough demonstrates how to encrypt and decrypt the contents a file. The code examples are designed for a Windows Forms application. This application does not demonstrate real-world scenarios, such as using smart cards. Instead, it demonstrates the fundamentals of encryption and decryption.
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:22]:
```suggestion
This walkthrough demonstrates how to encrypt and decrypt the contents of a file. The code examples are designed for a Windows Forms application. This application does not demonstrate real-world scenarios, such as using smart cards. Instead, it demonstrates the fundamentals of encryption and decryption.
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 2]
```diff
@@ -1,223 +1,203 @@
...
+- Use the <xref:System.Security.Cryptography.RSA> asymmetric algorithm to encrypt and decrypt the key to the data encrypted by <xref:System.Security.Cryptography.Aes>. Asymmetric algorithms are best used for smaller amounts of data, such as a key.
+
+  > [!NOTE]
+  > If you want to protect data on your computer instead of exchanging encrypted content with other people, consider using the <xref:System.Security.Cryptography.ProtectedData> class.
+
+The following table summarizes the cryptographic tasks in this topic.
+
+| Task | Description |
+|--|--|
+| Creating a Windows Forms application | Lists the controls that are required to run the application. |
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:36]:
```suggestion
| Create a Windows Forms application | Lists the controls that are required to run the application. |
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 3]
```diff
@@ -1,223 +1,203 @@
...
+| Declaring global objects | Declares string path variables, the <xref:System.Security.Cryptography.CspParameters>, and the <xref:System.Security.Cryptography.RSACryptoServiceProvider> to have global context of the <xref:System.Windows.Forms.Form> class. |
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:37]:
```suggestion
| Declaring global objects | Declares string path variables, the <xref:System.Security.Cryptography.CspParameters>, and the <xref:System.Security.Cryptography.RSACryptoServiceProvider> to have global context of the <xref:System.Windows.Forms.Form> class. |
```
```suggestion
| Declare global objects | Declares string path variables, the <xref:System.Security.Cryptography.CspParameters>, and the <xref:System.Security.Cryptography.RSACryptoServiceProvider> to have global context of the <xref:System.Windows.Forms.Form> class. |
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 4]
```diff
@@ -1,223 +1,203 @@
...
+| Creating an asymmetric key | Creates an asymmetric public and private key-value pair and assigns it a key container name. |
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:38]:
```suggestion
| Create an asymmetric key | Creates an asymmetric public and private key-value pair and assigns it a key container name. |
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 5]
```diff
@@ -1,223 +1,203 @@
...
+| Encrypting a file | Displays a dialog box to select a file for encryption and encrypts the file. |
+| Decrypting a file | Displays a dialog box to select an encrypted file for decryption and decrypts the file. |
+| Getting a private key | Gets the full key pair using the key container name. |
+| Exporting a public key | Saves the key to an XML file with only public parameters. |
+| Importing a public key | Loads the key from an XML file into the key container. |
+| Testing the application | Lists procedures for testing this application. |
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:44]:
```suggestion
| Encrypt a file | Displays a dialog box to select a file for encryption and encrypts the file. |
| Decrypt a file | Displays a dialog box to select an encrypted file for decryption and decrypts the file. |
| Get a private key | Gets the full key pair using the key container name. |
| Export a public key | Saves the key to an XML file with only public parameters. |
| Import a public key | Loads the key from an XML file into the key container. |
| Test the application | Lists procedures for testing this application. |
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 6]
```diff
@@ -1,223 +1,203 @@
...
+| <xref:System.Windows.Forms.Button>         | `buttonExportPublicKey`  | Export Public Key         |
+| <xref:System.Windows.Forms.Button>         | `buttonImportPublicKey`  | Import Public Key         |
+| <xref:System.Windows.Forms.Button>         | `buttonGetPrivateKey`    | Get Private Key           |
+| <xref:System.Windows.Forms.Label>          | `label1`                 | Key not set               |
+| <xref:System.Windows.Forms.OpenFileDialog> | `_encryptOpenFileDialog` |                           |
+| <xref:System.Windows.Forms.OpenFileDialog> | `_decryptOpenFileDialog` |                           |
+
+Double-click the buttons in the Visual Studio designer to create their event handlers.
+
+## Declaring global objects
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:70]:
```suggestion
## Declare global objects
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 7]
```diff
@@ -1,223 +1,203 @@
...
-
-This task saves the key created by the `Create Keys` button to a file. It exports only the public parameters.  
-  
-This task simulates the scenario of Alice giving Bob her public key so that he can encrypt files for her. He and others who have that public key will not be able to decrypt them because they do not have the full key pair with private parameters.  
-  
-Add the following code as the `Click` event handler for the `Export Public Key` button (`buttonExportPublicKey_Click`).  
-  
+[!code-vb[CryptoWalkThru#1](../../../samples/snippets/visualbasic/VS_Snippets_CLR/CryptoWalkThru/vb/Form1.vb#1)]
+
+## Creating an asymmetric key
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:77]:
```suggestion
## Create an asymmetric key
```

File: docs/standard/security/walkthrough-creating-a-cryptographic-application.md [hunk 8]
```diff
@@ -1,223 +1,203 @@
...
+
+This task creates an asymmetric key that encrypts and decrypts the <xref:System.Security.Cryptography.Aes> key. This key was used to encrypt the content and it displays the key container name on the label control.
+
+Add the following code as the `Click` event handler for the `Create Keys` button (`buttonCreateAsmKeys_Click`).
+
+[!code-csharp[CryptoWalkThru#2](../../../samples/snippets/csharp/VS_Snippets_CLR/CryptoWalkThru/cs/Form1.cs#2)]
+[!code-vb[CryptoWalkThru#2](../../../samples/snippets/visualbasic/VS_Snippets_CLR/CryptoWalkThru/vb/Form1.vb#2)]
+
+## Encrypting a file
```
Comment (tdykstra) [inline docs/standard/security/walkthrough-creating-a-cryptographic-application.md:86]:
```suggestion
## Encrypt a file
```

**Review by tdykstra — APPROVED** [review]
LGTM - Would it be possible to link to the completed project, or are the snippet tags too obtrusive for that to make sense?

**Comment by IEvangelist** [comment]
> LGTM - Would it be possible to link to the completed project, or are the snippet tags too obtrusive for that to make sense?

That is a good idea, I think we should link to samples browser bits.

**Pull request merged**